plugins/timesync.koplugin/main.lua

```diff
@@ -0,0 +1,57 @@
+
+local Device = require("device")
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local T = require("ffi/util").template
+local _ = require("gettext")
+
+if not (Device:isKobo() or Device:isKindle() or Device:isPocketbook()) then
+    return { disabled = true, }
+end
+
+local TimeSync = WidgetContainer:new{
+    name = "timesync",
+}
+
+local function currentTime()
+    local std_out = io.popen("date")
+    if std_out then
+        local result = std_out:read("*all")
+        std_out:close()
+        return T(_("New Time is %1"), result)
+    else
+        return _("Time synchronized")
```

Comment:
No, I have not got a nil. But according to Lua reference https://www.lua.org/manual/5.1/manual.html#pdf-io.popen, and some examples from google https://goo.gl/SykAui, it looks like, usually it's safe to add a ~= nil check here.

When this function is being executed, the synchronization has been finished already. Though we may not be able to show the correct time, as date command failed for any reason, showing "failed to synchronize time" seems not correct.